api/repositories/isolation_segments.go

```diff
@@ -0,0 +1,25 @@
+package repositories
+
+import (
+	"time"
+)
+
+type IsolationSegmentRepo struct {
+	klient Klient
+}
+
+func NewIsolationSegmentRepo(
```

Comment:
Does a repo without methods make sense? I also do not see why we need the record type as well.